src/v/cluster/rm_stm.cc

```diff
@@ -1594,13 +1594,22 @@ ss::future<result<kafka_result>> rm_stm::replicate_seq(
         }
         auto [seq_it, inserted] = _log_state.seq_table.try_emplace(bid.pid);
         if (inserted) {
-            seq_it->second.pid = bid.pid;
-            seq_it->second.seq = front->last_seq;
-            seq_it->second.last_offset = front->r.value().last_offset;
+            seq_it->second.entry.pid = bid.pid;
+            seq_it->second.entry.seq = front->last_seq;
+            seq_it->second.entry.last_offset = front->r.value().last_offset;
         } else {
-            seq_it->second.update(
+            seq_it->second.entry.update(
               front->last_seq, front->r.value().last_offset);
         }
+
+        if (!bid.is_transactional) {
+            if (!inserted) {
+                _log_state.lru_idempotent_pids.erase(
+                  _log_state.lru_idempotent_pids.iterator_to(seq_it->second));
+            }
+            _log_state.lru_idempotent_pids.push_back(seq_it->second);
+        }
```

Comment:
can you leave a comment here explaining why `inserted` (which doesn't seem to be related directly to the state of the lru list--just `seq_table`) should control if the the entry is on the lru list or not already. to make this code simpler, you might consider inspecting the lru entry itself: if the entry is on a list, remove it first.